### PR TITLE
[query/service] fix fs-related issues

### PR DESCRIPTION
**Pull request by danking**
Do not approve this if test_backwards_compatibility dramatically inflates test time

### CODE REVIEW

File: hail/python/hail/fs/router_fs.py [hunk 1]
```diff
@@ -230,10 +238,9 @@ async def size():
 
     def ls(self, path: str, _max_simultaneous_files: int = 50) -> List[Dict]:
         async def _ls():
-            async with OnlineBoundedGather2(asyncio.Semaphore(_max_simultaneous_files)) as pool:
-                tasks = [pool.call(self._fle_to_dict, fle)
-                         async for fle in await self.afs.listfiles(path)]
-                return [await t for t in tasks]
+            return await bounded_gather(
+                *[functools.partial(self._fle_to_dict, fle)
```
Comment (jigold) [inline hail/python/hail/fs/router_fs.py:242]: Do we not care how many simultaneous queries we do?

Comment (danking) [reply hail/python/hail/fs/router_fs.py:242]: Ah, yes, I dropped the parallelism argument. I restored it

**Comment by johnc1231** [comment]
Has conflict

**Comment by danking** [comment]
should work now

**Comment by danking** [comment]
@jigold since you already took a look I sent back to you. It passed tests on last run other than the MovieLens issue.

**Comment by jigold** [comment]
Can you give me some context for this change? It seems fine to me, but I don't see how it was different than what was there before.

**Comment by jigold** [comment]
Can you give me some context for this change? It seems fine to me, but I don't see how it was different than what was there before.

**Comment by danking** [comment]
`AsyncFS.exist(url)` is true if and only if `url` is a blob (aka a file). Hail Query's `FS` file system API mimics Hadoop's file system API which mimics a UNIX-like file system api, so we gotta check for both isdir and isfile.

**Comment by jigold** [comment]
Great! Can you check whether the issue with backwards compatibility you were worried about is indeed a problem?

**Comment by danking** [comment]
There's another bug which means that test still fails and fails in ~0 seconds. I'm gonna put off fixing that bug for now because I'm not sure what to do about that test. It's currently marked as fails_service_backend.

**Comment by danking** [comment]
merged into https://github.com/hail-is/hail/pull/11599 because I need both fixes together to pass all the tests.

**Pull request closed without merging**